planning/behavior_path_planner/src/scene_module/avoidance/avoidance_utils.cpp

```diff
@@ -559,4 +704,63 @@ lanelet::ConstLanelets getTargetLanelets(
   return target_lanelets;
 }
 
+boost::optional<double> calcDecelDistWithJerkAndAccConstraints(
+  const double current_vel, const double target_vel, const double current_acc, const double acc_min,
+  const double jerk_acc, const double jerk_dec)
+{
+  constexpr double epsilon = 1e-3;
+  const double t_dec =
+    acc_min < current_acc ? (acc_min - current_acc) / jerk_dec : (acc_min - current_acc) / jerk_acc;
+  const double t_acc = (0.0 - acc_min) / jerk_acc;
+  const double t_min = (target_vel - current_vel - current_acc * t_dec -
+                        0.5 * jerk_dec * t_dec * t_dec - 0.5 * acc_min * t_acc) /
+                       acc_min;
+
+  // check if it is possible to decelerate to the target velocity
+  // by simply bringing the current acceleration to zero.
+  const auto is_decel_needed =
+    0.5 * (0.0 - current_acc) / jerk_acc * current_acc > target_vel - current_vel;
+
+  if (t_min > epsilon) {
+    return calcDecelDistPlanType1(
```

Comment:
Type 1/2/3 is super vague, nobody can understand the implementation. Please write an explanation in header or somewhere.

Reply:
Thanks for your comment. I added description about `caclDecelDistPlanType*` in https://github.com/autowarefoundation/autoware.universe/pull/2515/commits/904527e90deb9b30cb5fcdc8c9220725f441e849.